proto/iotics/api/meta.proto

```diff
@@ -168,3 +143,58 @@ message SparqlUpdateResponse {
   // SPARQL update response headers
   Headers headers = 1;
 }
+
+// FKGQueryRequest describes a SPARQL query.
+message FKGQueryRequest {
+
+  // SPARQL query request payload.
+  message Payload {
+
+    // The desired result content type. Note that choosing an invalid result type for the type of query will result in
+    // an error status reported in the response. (See SparqlResultType for valid content-query type combinations.)
+    SparqlResultType resultContentType = 1;
+    // A UTF8-encoded SPARQL 1.1 query
+    bytes query = 2;
+  }
+
+  // SPARQL query request headers
+  Headers headers = 1;
+
+  // SPARQL query request scope
+  Scope scope = 2;
+
+  // SPARQL query request payload
+  Payload payload = 3;
+}
+
+// FKGQueryResponse is a part of a result for a SPARQL query request. Multiple chunks form a complete result. Related
+// chunks can be identified by a combination of:
+// - Client reference (in headers, set by caller)
+// - Chunk sequence number
+message FKGQueryResponse {
+
+  // Payload of the query result chunk
+  message Payload {
+
+    // Position of a chunk in result for a given request. The first chunk has a sequence number of 0.
+    uint64 seqNum = 1;
+
+    // Indicates whether this is the last chunk, for a specific request. Results for different requests can be
+    // identified by setting a unique clientRef in the request headers.
+    bool last = 2;
+
+    // Content type of the result.
+    SparqlResultType contentType = 3;
```

Comment:
do you need this? or will it be == to the valye in the request

Reply:
This is so that the receiver of responses doesn't need to know about the request necessarily. (We have the same approach in the original SparqlQuery and other fields such as - headers with client references)